### PR TITLE
Add `root` and `log` filters.

### PR DESCRIPTION
**Pull request by mattcoley**
Adds a `root` and a `log` filter as requested here https://github.com/HubSpot/jinjava/issues/292 similar to https://ansible-docs.readthedocs.io/zh/stable-2.0/rst/playbooks_filters.html#math.

Complications: supporting BigDecimal and exact values requires adding a dependency on https://github.com/eobermuhlner/big-math

### CODE REVIEW

File: src/main/java/com/hubspot/jinjava/lib/filter/LogFilter.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+import com.hubspot.jinjava.doc.annotations.JinjavaSnippet;
+import com.hubspot.jinjava.interpret.InvalidArgumentException;
+import com.hubspot.jinjava.interpret.InvalidInputException;
+import com.hubspot.jinjava.interpret.InvalidReason;
+import com.hubspot.jinjava.interpret.JinjavaInterpreter;
+
+import ch.obermuhlner.math.big.BigDecimalMath;
+
+@JinjavaDoc(
+    value = "Return the log of the input.",
```
Comment (jboulter) [inline src/main/java/com/hubspot/jinjava/lib/filter/LogFilter.java]: "return the natural log value of the input"

File: src/test/java/com/hubspot/jinjava/lib/filter/RootFilterTest.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  }
+
+  @Test
+  public void itCalculatesNthRoot() {
+    assertThat(filter.filter(9765625d, interpreter, "5.0")).isEqualTo(25d);
+  }
+
+  @Test
+  public void itCalculatesNthRootOfBigDecimal() {
+    BigDecimal result = ((BigDecimal) filter.filter(new BigDecimal(9765625d), interpreter, "5.0"));
```
Comment (jboulter) [inline src/test/java/com/hubspot/jinjava/lib/filter/RootFilterTest.java]: can this be computationally expensive with large values?

File: src/main/java/com/hubspot/jinjava/lib/filter/LogFilter.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    }
+
+    if (base != null && base <= 0) {
+      throw new InvalidArgumentException(interpreter, this, InvalidReason.POSITIVE_NUMBER, 0, base);
+    }
+  }
+
+  @Override
+  public String getName() {
+    return "log";
```
Comment (jboulter) [inline src/main/java/com/hubspot/jinjava/lib/filter/LogFilter.java]: I'm not thrilled about the very generic name of this, but it makes sense to be compatible with ansible.

File: src/test/java/com/hubspot/jinjava/lib/filter/RootFilterTest.java (hunk 2 above)
Comment (mattcoley) [reply src/test/java/com/hubspot/jinjava/lib/filter/RootFilterTest.java]: Performance is largely a function of precision in this case set to 50 digits. With such a low precision operations are in the nanoseconds range. The goal of this library we are using is for performance remain stable as the input grows so we should be safe http://obermuhlner.ch/wordpress/2016/06/02/bigdecimalmath/

**Pull request merged**